src/libraries/System.Numerics.Vectors/tests/GenericVectorTests.cs

```diff
@@ -846,6 +846,72 @@ public void VectorSingleEqualsNaNTest()
             var nan = new Vector<float>(float.NaN);
             Assert.True(nan.Equals(nan));
         }
+
+        [Fact]
+        public void VectorDoubleEqualsNonCanonicalNaNTest()
+        {
+            var cast = BitConverter.DoubleToUInt64Bits(double.NaN);
+            var one = BitConverter.DoubleToUInt64Bits(1);
+            var sign = one ^ BitConverter.DoubleToUInt64Bits(-1);
+            var mant = one ^ BitConverter.DoubleToUInt64Bits(1.2);
```

Comment:
`0xF_FFFF_FFFF_FFFF` might be clearer here as well if you wanted just the mantissa bits.

Reply:
using (1UL << 53) - 1 as that seemed easier to understand.